src/Cli/dotnet/PerformanceLogEventSource.cs

```diff
@@ -0,0 +1,341 @@
+        }
+
+        [Event(12)]
+        internal void TelemetrySendIfEnabledStop()
+        {
+            WriteEvent(12);
+        }
+
+        [Event(13)]
+        internal void BuiltInCommandStart()
+        {
+            WriteEvent(13);
+        }
+
+        [Event(14)]
+        internal void BuiltInCommandStop()
+        {
+            WriteEvent(14);
+        }
+
+        [Event(15)]
+        internal void BuiltInCommandParserStart()
+        {
+            WriteEvent(15);
+        }
+
+        [Event(16)]
+        internal void BuiltInCommandParserStop()
+        {
+            WriteEvent(16);
+        }
+
+        [Event(17)]
+        internal void ExtensibleCommandResolverStart()
+        {
+            WriteEvent(17);
+        }
+
+        [Event(18)]
+        internal void ExtensibleCommandResolverStop()
+        {
+            WriteEvent(18);
+        }
+
+        [Event(19)]
+        internal void ExtensibleCommandStart()
+        {
+            WriteEvent(19);
+        }
+
+        [Event(20)]
+        internal void ExtensibleCommandStop()
+        {
+            WriteEvent(20);
+        }
+
+        [Event(21)]
+        internal void TelemetryClientFlushStart()
+        {
+            WriteEvent(21);
+        }
+
+        [Event(22)]
+        internal void TelemetryClientFlushStop()
+        {
+            WriteEvent(22);
+        }
+
+        [NonEvent]
+        internal void LogMachineConfiguration()
+        {
+            if(IsEnabled())
+            {
+                MachineConfiguration(Environment.MachineName, Environment.ProcessorCount);
+            }
+        }
+
+        [Event(23)]
+        internal void MachineConfiguration(string machineName, int processorCount)
+        {
+            WriteEvent(23, machineName, processorCount);
+        }
+
+        [NonEvent]
+        internal void LogDrives()
+        {
+            if (IsEnabled())
+            {
+                foreach (DriveInfo driveInfo in DriveInfo.GetDrives())
+                {
+                    try
+                    {
+                        DriveConfiguration(driveInfo.Name, driveInfo.DriveFormat, driveInfo.DriveType.ToString(), (double)driveInfo.TotalSize/1024/1024, (double)driveInfo.AvailableFreeSpace/1024/1024);
+                    }
+                    catch
+                    {
+                    }
+                }
+            }
+        }
+
+        [Event(24)]
+        internal void DriveConfiguration(string name, string format, string type, double totalSizeMB, double availableFreeSpaceMB)
+        {
+            WriteEvent(24, name, format, type, totalSizeMB, availableFreeSpaceMB);
+        }
+
+        [Event(25)]
+        internal void AssemblyLoad(string assemblyName, double timeInMs)
+        {
+            WriteEvent(25, assemblyName, timeInMs);
+        }
+
+        [NonEvent]
+        internal void LogMemoryConfiguration()
+        {
+            if(RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
only works on windows? any plan on linux and mac?

Reply:
Yes, the code as is only works on Windows.  I left it at this for the prototype, but it does look like we could crib some code from the PAL to implement this for Linux: [https://github.com/dotnet/runtime/blob/fd094a92cdcd7ccf3ae1efad5922b9a303bf44b9/src/coreclr/src/pal/src/misc/sysinfo.cpp#L323](https://github.com/dotnet/runtime/blob/fd094a92cdcd7ccf3ae1efad5922b9a303bf44b9/src/coreclr/src/pal/src/misc/sysinfo.cpp#L323).  It's certainly more complicated though.

---
In reply to: [505711118](https://github.com/wli3/sdk/pull/1#discussion_r505711118) [](ancestors = 505711118)